File: binary-search/BinarySearch.java
import java.util.Scanner;

public class BinarySearch {

    private int[] array;


    public static void main(String[] args) {

        BinarySearch search = new BinarySearch();
        Scanner scanner = new Scanner(System.in);

        int N = Integer.parseInt(scanner.nextLine());
        String[] inputArray = (scanner.nextLine()).split(" ");

        search.array = new int[N];
        for (int i = 0; i < N; i++) {
            search.array[i] = Integer.parseInt(inputArray[i]);
        }

        int K = Integer.parseInt(scanner.nextLine());
        for (int i = 0; i < K; i++) {
            System.out.println(search.binSearch(Integer.parseInt(scanner.nextLine())));
        }

    }

    public int binSearch(int number) {

        int left = 0;
        int right = array.length - 1;
        int middle;

        while (true) {

            if (left > right)
                return -1;

            middle = (left + right) / 2;
            if (array[middle] == number)
                return middle;

            if (array[middle] > number) {
                right = middle - 1;
            } else {
                left = middle + 1;
            }

        }
    }
}

File: binary-search/binary_search.cpp
#include <iostream>

using namespace std;

int binSearch(long number, long* Arr, int size) {
    int result = -1;
    int left, right, middle;
    left = 0;
    right = size - 1;
    while (true) {
        if (right - left <= 1) {
            if (number == Arr[left]) {
                result = left;
            }
            else if (number == Arr[right]) {
                result = right;
            }
            break;
        }
        middle = (left + right)/2;
        if (number == Arr[middle]) {
            result = middle;
            break;
        }
        else if (number > Arr[middle])
        {
            left = middle;
        }
        else {
            right = middle;
        }
    }

    return result;
}

int main() {
    int N;
    cin >> N;

    long Arr[N];
    for (int i = 0; i < N; i++) {
        cin >> Arr[i];
    }

    int K;
    cin >> K;

    long x_i[K];
    for (int i = 0; i < K; i++) {
        cin >> x_i[i];
    }

    for (int i = 0; i < K; i++) {
        cout << binSearch(x_i[i], Arr, N) << endl;
    }

    return 0;
}
